src/frequenz/client/base/channel.py

```diff
@@ -41,6 +42,24 @@ class SslOptions:
     """
 
 
+@dataclasses.dataclass(frozen=True)
+class KeepAliveOptions:
+    """Options for HTTP2 keep-alive pings."""
+
+    enabled: bool = True
+    """Whether keep-alive should be enabled."""
+
+    interval: timedelta | None = None
+    """The interval between pings.
+
+    If None, keep-alive is disabled."""
```

Comment:
I would not allow `None` here, and provide some default, as `enabled` already covers for the disabled case. Otherwise having `enabled=True, interval=None` doesn't make much sense (also the current default is weird, it is `enabled=True` but it is not actually enabled.

For example:
```suggestion
    interval: timedelta = timedelta(seconds=5)
    """The interval between pings."""
```

Reply:
There may not be a default keep-alive that works for the general case, but 5s seems excessive.  Nginx waits for 75s, so I'll set it to 60s: http://nginx.org/en/docs/http/ngx_http_core_module.html#keepalive_timeout

Reply:
> also the current default is weird, it is `enabled=True` but it is not actually enabled.

:D true, I didn't think of that.

Reply:
done